doc/java-generator.md

```diff
@@ -0,0 +1,85 @@
+# Java generation (preview)
```

Comment:
Maybe rename it `Java generation from CRD`.
Also I would be consistent with the file name `java-generation-from-crd.md`

Reply:
ok